storm-core/test/jvm/org/apache/storm/utils/UtilsTest.java

```diff
@@ -0,0 +1,221 @@
+                System.clearProperty(key);
+            } else {
+                System.setProperty(key, oldValue);
+            }
+        }
+    }
+
+    private CuratorFrameworkFactory.Builder setupBuilder(boolean withExhibitor){
+        return setupBuilder(withExhibitor, false /*without Auth*/);
+    }
+
+    private CuratorFrameworkFactory.Builder setupBuilder(boolean withExhibitor, boolean withAuth){
+        CuratorFrameworkFactory.Builder builder = CuratorFrameworkFactory.builder();
+        Map<String, Object> conf = new HashMap<String, Object>();
+        if (withExhibitor){
+            conf.put(Config.STORM_EXHIBITOR_SERVERS,"foo");
+            conf.put(Config.STORM_EXHIBITOR_PORT, 0);
+            conf.put(Config.STORM_EXHIBITOR_URIPATH, "/exhibitor");
+            conf.put(Config.STORM_EXHIBITOR_POLL, 0);
+            conf.put(Config.STORM_EXHIBITOR_RETRY_INTERVAL, 0);
+            conf.put(Config.STORM_EXHIBITOR_RETRY_INTERVAL_CEILING, 0);
+            conf.put(Config.STORM_EXHIBITOR_RETRY_TIMES, 0);
+        }
+        conf.put(Config.STORM_ZOOKEEPER_CONNECTION_TIMEOUT, 0);
+        conf.put(Config.STORM_ZOOKEEPER_SESSION_TIMEOUT, 0);
+        conf.put(Config.STORM_ZOOKEEPER_RETRY_INTERVAL, 0);
+        conf.put(Config.STORM_ZOOKEEPER_RETRY_INTERVAL_CEILING, 0);
+        conf.put(Config.STORM_ZOOKEEPER_RETRY_TIMES, 0);
+        String zkStr = new String("zk_connection_string");
+        ZookeeperAuthInfo auth = null;
+        if (withAuth){
+            auth = new ZookeeperAuthInfo("scheme", "abc".getBytes());
+        }
+        Utils.testSetupBuilder(builder, zkStr, conf, auth);
+        return builder;
+    }
+
+    @Test
+    public void ifExhibitorServersProvidedBuilderUsesTheExhibitorEnsembleProviderTest(){
+        CuratorFrameworkFactory.Builder builder = setupBuilder(true /*with exhibitor*/);
+        Assert.assertEquals(builder.getEnsembleProvider().getConnectionString(), "");
+        Assert.assertEquals(builder.getEnsembleProvider().getClass(), ExhibitorEnsembleProvider.class);
+    }
+
+    @Test
+    public void ifExhibitorServersAreEmptyBuilderUsesAFixedEnsembleProviderTest(){
+        CuratorFrameworkFactory.Builder builder = setupBuilder(false /*without exhibitor*/);
+        Assert.assertEquals(builder.getEnsembleProvider().getConnectionString(), "zk_connection_string");
+        Assert.assertEquals(builder.getEnsembleProvider().getClass(), FixedEnsembleProvider.class);
+    }
+
+    @Test
+    public void ifAuthSchemeAndPayloadAreDefinedBuilderUsesAuthTest(){
+        CuratorFrameworkFactory.Builder builder = setupBuilder(false /*without exhibitor*/, true /*with auth*/);
+        List<AuthInfo> authInfos = builder.getAuthInfos();
+        AuthInfo authInfo = authInfos.get(0);
+        Assert.assertEquals(authInfo.getScheme(), "scheme"); 
+        Assert.assertArrayEquals(authInfo.getAuth(), "abc".getBytes());
+    }
+
+    @Test
+    public void parseJvmHeapMemByChildOpts1024KIs1Test(){
```

Comment:
Can we have all the asserts in a single test case. These seem to be testing one specific functionality